cf-java-logging-support-servlet/src/main/java/com/sap/hcp/cf/logging/servlet/filter/CorrelationIdFilter.java

```diff
@@ -59,29 +56,31 @@ private String determineCorrelationId(HttpServletRequest request) {
             correlationId = String.valueOf(UUID.randomUUID());
             // add correlation-id as custom field, since it is added to MDC only
             // in the next step
-            LOG.debug("Generated new correlation-id <{}>", correlationId, customField(correlationHeader.getField(),
-                                                                                      correlationId));
+            LOG.debug("Generated new correlation-id <{}>", correlationId);
         }
         return correlationId;
     }
 
-    private boolean isBlankOrDefault(String value) {
+    private static boolean isBlankOrDefault(String value) {
         return value == null || value.isEmpty() || value.equals(Defaults.UNKNOWN);
     }
 
     private String getCorrelationIdFromTraceparent(HttpServletRequest request) {
         String traceparent = HttpHeaderUtilities.getHeaderValue(request, traceparentHeader);
-        return ofNullable(traceparent).filter(not(this::isBlankOrDefault)).map(this::parseTraceparent).orElse(
-                                                                                                                        null);
+        return isBlankOrDefault(traceparent) ? null : parseTraceparent(traceparent);
     }
 
-    private <T> Predicate<T> not(Predicate<T> p) {
-        return p.negate();
-    }
-
-    private String parseTraceparent(String value) {
-        String[] tokens = value.split("-");
-        return tokens.length >= 2 ? tokens[1] : null;
+    private static String parseTraceparent(String value) {
+        int idx1 = value.indexOf('-');
+        if (idx1 != -1) {
+            ++idx1;
+            int idx2 = value.indexOf('-', idx1);
+            if (idx2 != -1) {
+                // return string between first and second '-'.
+                return value.substring(idx1, idx2);
+            }
+        }
+        return null;
```

Comment:
I can see the allocation benefit of this approach. Is calling `indexOf` multiple times indeed faster than the original approach?

Reply:
The original approach creates at least 3 Strings (maybe more, depending on the `value`) and an additional String array and it scans until the end to populate the array.
The proposed approach bounds the allocation to max. one string and scans no further that to the second '-'.

I have less worries about the scanning performance but more regarding the superfluous allocations.